n3fit/src/n3fit/layers/observable.py

```diff
@@ -164,7 +169,7 @@ def call(self, pdf):
 
         observables = []
         for idx, (pdf, padded_fk) in enumerate(zip(pdfs, self.padded_fk_tables)):
-            pdf_to_convolute = pdf if self.boundary_pdf[idx] is None else self.boundary_pdf[idx]
+            pdf_to_convolute = [pdf if p is None else p for p in self.boundary_pdf[idx]]
```

Comment:
Yes, there are (will be) situations in which we have multiple boundary conditions. So this way we are as general as possible, this can even already accommodate for Fragmentation Function fits (which we'll be able to do very soon).

Reply:
Then the `determine_pdf` function should be in use also here. That way adding FF means just adding them in that function (instead of adding them differently in different parts of the code).

It also makes it easier to eventually lift the function to be "something else" (maybe another PDF, or vp object coming from n3fit, etc)